pages/programs/highschool/apply.tsx

```diff
@@ -10,8 +10,32 @@ import { BgImg } from '@this/src/components/Elements';
 import Map from '@this/src/components/Elements/Map';
 import HighschoolApplicationForm from '@this/src/Forms/Form.HighschoolApplication';
 import PlainCard from '@this/src/components/Cards/PlainCard';
+import { courseTimes } from '@this/src/Forms/formData/highSchoolApplicationData';
+import { IHighschoolPrograms } from '@this/data/types/programs';
+import { InfoIcon } from '@this/src/components/icons/Info';
+import { getStaticAsset } from '@this/pages-api/static/[asset]';
+import { useState } from 'react';
+import { useClickAway } from '@this/src/hooks/useClickAway';
+
+type HighschoolSignupProps = {
+  courses: IHighschoolPrograms['courses'];
+};
+
+const HighschoolSignup: NextPage<HighschoolSignupProps> = ({ courses }) => {
+  const [courseInfo, setCourseInfo] = useState<IHighschoolPrograms['courses'][0] | null>(null);
+  const [courseInfoRef, isOpen, setOpen] = useClickAway(() => setCourseInfo(null));
+
+  const toggleCourseInfo = (course?: string) => {
+    const c = courses.find(({ id }) => course === id);
+    if (!course || !c || c.id === courseInfo?.id) {
+      setCourseInfo(null);
+      return setOpen(false);
+    }
+
+    setCourseInfo(c);
+    setOpen(true);
+  };
```

Comment:
Just a suggestion, but I think it worth the few extra lines to break up the condition on 30
```suggestion
const toggleCourseInfo = (course?: string) => {
  if (!course || !courseInfo) {
    return setOpen(false);
  }
  const c = courses.find(({ id }) => course === id);
  if (!c || c.id === courseInfo.id) {
    setCourseInfo(null);
    return setOpen(false);
  }

  setCourseInfo(c);
  setOpen(true);
};

```